src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -875,25 +876,18 @@ public ImmutableArray<T> RemoveAll(Predicate<T> match)
             Requires.NotNull(match, "match");
 
             if (self.IsEmpty)
-            {
-                return new ImmutableArray<T>(self.array);
-            }
+                return self;
 
-            List<int> removeIndexes = null;
+            List<int> removeIndexes = new List<int>();
```

Comment:
In the event no item is removed, this code allocates a `List<int>` instance which is never used. I believe this change deserves a much closer look during the code review, which is why I pointed it out.